File: client/src/app/shared/models/Review.ts
export class Review {
    id: number = 0;
    body: string = '';
    createdAt: Date = new Date();
    authorName: string = '';
    authorId: number = 0;
    productId: number = 0;
    rating: number = 0;

    constructor(body?: string, productId?: number) {
        if (body && productId) {
            this.body = body;
            this.productId = productId;
        }
    }
}

File: API/Models/Review.cs
using System;
using System.Text.Json.Serialization;

namespace API.Models
{
    public class Review : BaseEntity
    {
        public int AuthorId { get; set; }
        public string AuthorName { get; set; }
        public string Body { get; set; }
        public DateTime CreatedAt { get; set; }
        public int Rating { get; set; }
        public int ProductId { get; set; }
        [JsonIgnore] public Product Product { get; set; }
    }
}